src/main/java/org/tugraz/sysds/hops/recompile/Recompiler.java

```diff
@@ -396,12 +396,20 @@ else if( !codegen ) {
 	
 	private static void logExplainDAG(StatementBlock sb, ArrayList<Hop> hops, ArrayList<Instruction> inst) {
 		if( DMLScript.EXPLAIN == ExplainType.RECOMPILE_HOPS ) {
-			System.out.println("EXPLAIN RECOMPILE \nGENERIC (lines "+sb.getBeginLine()+"-"+sb.getEndLine()+"):\n" +
-				Explain.explainHops(hops, 1));
+			if (sb != null)
+				System.out.println("EXPLAIN RECOMPILE \nGENERIC (lines "+sb.getBeginLine()+"-"+sb.getEndLine()+"):\n" +
+						Explain.explainHops(hops, 1));
+			else
+				System.out.println("EXPLAIN RECOMPILE \nGENERIC (lines "+hops.get(0).getBeginLine()+"-"+hops.get(0).getEndLine()+"):\n" +
+						Explain.explainHops(hops, 1));
```

Comment:
Try to avoid this code duplication - both sb and hop implement ParseInfo - so you can extract this once (with a conditional ternary operator) and then simply use this ParseInfo to get the line numbers.